yjit_codegen.c

```diff
@@ -1892,17 +1900,14 @@ gen_opt_getinlinecache(jitstate_t *jit, ctx_t *ctx)
 
     // See vm_ic_hit_p().
     struct iseq_inline_constant_cache_entry *ice = ic->entry;
-    if (!ice) {
-        // Cache not filled
-        return YJIT_CANT_COMPILE;
-    }
-    if (ice->ic_serial != ruby_vm_global_constant_state) {
-        // Cache miss at compile time.
-        return YJIT_CANT_COMPILE;
-    }
-    if (ice->ic_cref) {
-        // Only compile for caches that don't care about lexical scope.
-        return YJIT_CANT_COMPILE;
+    if (!ice || // cache not filled
+        ice->ic_serial != ruby_vm_global_constant_state || // cache out of date
+        ice->ic_cref /* cache only valid for certain lexical scopes */) {
+        // In these cases, leave a block that unconditionally side exits
+        // for the interpreter to invalidate.
+        uint8_t *side_exit = yjit_side_exit(jit, ctx);
+        jmp_ptr(cb, side_exit);
+        return YJIT_END_BLOCK;
```

Comment:
The changes you made should make it possible to just return `YJIT_CANT_COMPILE` here?

Reply:
Still unclear on this.